pkg/runtime/wfengine/workflow.go

```diff
@@ -184,20 +196,54 @@ func (wf *workflowActor) createWorkflowInstance(ctx context.Context, actorID str
 		}
 	}
 
-	// We block (re)creation of existing workflows unless they are in a completed state
-	// Or if they still have any pending activity result awaited.
-	if !created {
-		runtimeState := getRuntimeState(actorID, state)
-		if !runtimeState.IsCompleted() {
-			return fmt.Errorf("an active workflow with ID '%s' already exists", actorID)
-		}
-		if wf.activityResultAwaited.Load() {
-			return fmt.Errorf("a terminated workflow with ID '%s' is already awaiting an activity result", actorID)
+	// orchestration doesn't exist, create it.
```

Comment:
Nit:
```suggestion
	// orchestration didn't exist and was just created
```

Reply:
update a bit please check again if it makes sense. Thank you!